internal/worker/sshserver/proxy.go

```diff
@@ -0,0 +1,47 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package sshserver
+
+import (
+	"time"
+
+	"github.com/juju/errors"
+
+	k8sexec "github.com/juju/juju/caas/kubernetes/provider/exec"
+	"github.com/juju/juju/core/virtualhostname"
+	"github.com/juju/juju/internal/worker/sshserver/k8s"
+	"github.com/juju/juju/internal/worker/sshserver/machine"
+)
+
+type proxyFactory struct {
+	k8sResolver k8s.Resolver
+	logger      Logger
+	connector   machine.SSHConnector
+}
+
+type connectionInfo struct {
+	startTime   time.Time
+	destination virtualhostname.Info
+}
+
+// New creates a new set of proxy handlers based on the destination type.
+func (b proxyFactory) New(info connectionInfo) (ProxyHandlers, error) {
```

Comment:
does this need to be exported? if yes, then make connectionInfo type exported as well..

Reply:
Good catch, it doesn't need to be exported.

Reply:
Ah but it's exported to satisfy the `ProxyFactory` interface and I think it would be weird to not export that interface or to make that an exported interface with a non-exported method.
So, in the end I will export `connectionInfo`.
I could go the other direction and choose not to export the interfaces we use for testing but that seems wrong, especially if we were to split up this package (which I think we should do at some point to make the worker just handle dependency injection and put the SSH server in its own package).